src/lib/db-types.ts

```diff
@@ -20,26 +20,50 @@ export interface DBDocumentMap {
   [key /* documentKey */: string]: DBDocument;
 }
 
-export type DBDocumentType = "section" | "learningLog";
+export type DBDocumentType = DBUserDocumentType | DBOfferingDocumentType;
+export type DBUserDocumentType = "section" | "learningLog";
+export type DBOfferingDocumentType = "published";
 
-export type DBDocumentMetadata = DBSectionDocumentMetadata | DBLearningLogDocumentMetadata;
-export interface DBDocumentMetadataBase {
+export interface DBDocumentMetadata {
   version: "1.0";
   self: {
-    uid: string;
     documentKey: string;
   };
   createdAt: number;
+  type: DBDocumentType;
```

Comment:
I also moved `type` into the `DBDocumentMetadata` class. This simplified document creation and allowed me to get rid of the `DBDocumentMetadata` type union. Let me know if there's a reason to move it back out.